packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -85,13 +88,14 @@ const BLOCK_EDITOR_SETTINGS = [
 /**
  * React hook used to compute the block editor settings to use for the post editor.
  *
- * @param {Object} settings EditorProvider settings prop.
- * @param {string} postType Editor root level post type.
- * @param {string} postId   Editor root level post ID.
+ * @param {Object} settings        EditorProvider settings prop.
+ * @param {string} postType        Editor root level post type.
+ * @param {string} postId          Editor root level post ID.
+ * @param {string} contextPostType The post type of the edited post.
  *
  * @return {Object} Block Editor Settings.
  */
-function useBlockEditorSettings( settings, postType, postId ) {
+function useBlockEditorSettings( settings, postType, postId, contextPostType ) {
```

Comment:
Copied from #60609.

> I think ideally the contextPostType should replace the postType argument, and we could pass a "mode" in addition. That way we can fallback to `wp_template` post type if the mode is "template-locked".

Reply:
@youknowriad I would rather change the signature to 
```suggestion
function useBlockEditorSettings( settings, post, template ) {
```

which to me makes more sense as we need both data to infer the proper settings.

Reply:
template is not necessarily defined though but I'd be fine with that too for now.

That said, I want to share that the fact that `EditorProvider` receives "objects" rather than "id" (for both post and template) is a mistake. It makes the whole thing re-render way too much. Components should receive granular props as much as possible. That said, now, it's a historic mistake that I'm hoping to reconsider at some point but there are some BC aspects to it.

Reply:
I guess in that case it'd be unwise to pass them around even more. Hmm,